automation/jleechanorg_pr_automation/tests/test_pr_filtering_matrix.py

```
@@ -0,0 +1,470 @@
+class TestPRFilteringMatrix(unittest.TestCase):
+    """Matrix testing for PR filtering and actionable counting logic"""
+
+    def setUp(self):
+        """Set up test environment"""
+        self.temp_dir = tempfile.mkdtemp()
+        self.monitor = JleechanorgPRMonitor()
+        self.monitor.history_storage_path = self.temp_dir
+
+    def tearDown(self):
+        """Clean up test files"""
+        import shutil
```

Ruff (0.14.5)

`import` should be at the top-level of a file

(PLC0415)